### PR TITLE
feat(pie-checkbox): DSW-2280 use classmap for styling

### PR DESCRIPTION
**Pull request by jamieomaguire**
## Describe your changes (can list changeset entries if preferable)
---
"@justeattakeaway/pie-checkbox": minor
---

[Changed] - use ClassMap for styling with CSS classes rather than data attributes

## Author Checklist (complete before requesting a review)
- [x] I have performed a self-review of my code
- [x] I have added thorough tests where applicable (unit / component / visual)
- [x] I have reviewed the `PIE Storybook`/`PIE Docs` PR preview
- [x] I have reviewed visual test updates properly before approving

### Reviewer 1 - @xander-marjoram 
- [x] I have reviewed the `PIE Storybook` PR preview

### Reviewer 2
- [x] I have reviewed the `PIE Storybook` PR preview

### CODE REVIEW

File: packages/components/pie-checkbox/test/component/pie-checkbox.spec.ts [hunk 1]
```diff
@@ -330,45 +329,6 @@ test.describe('PieCheckbox - Component tests', () => {
                 expect(assistiveText).not.toBeVisible();
             });
 
-            test('should apply the "default" variant attribute if no status is provided', async ({ mount, page }) => {
```
Comment (jamieomaguire) [inline packages/components/pie-checkbox/test/component/pie-checkbox.spec.ts:333]: These are covered by visual tests

File: packages/components/pie-checkbox/src/index.ts [hunk 2]
```diff
@@ -164,13 +165,26 @@ export class PieCheckbox extends FormControlMixin(RtlMixin(LitElement)) implemen
...
+            'c-checkbox': true,
+            [`c-checkbox--status-${status}`]: !componentDisabled && status,
+            'c-checkbox--disabled': componentDisabled,
+            'c-checkbox--checked': checked,
+            'c-checkbox--indeterminate': indeterminate && !checked,
+        };
+
+        const labelClasses = {
+            'c-checkbox-tick': true,
+            [`c-checkbox-tick--status-${status}`]: !componentDisabled && status,
```
Comment (xander-marjoram) [inline packages/components/pie-checkbox/src/index.ts:178]: `status` will always be truthy, I believe?
```suggestion
            [`c-checkbox-tick--status-${status}`]: !componentDisabled,
```

File: packages/components/pie-checkbox/src/index.ts [hunk 3]
```diff
@@ -164,13 +165,26 @@ export class PieCheckbox extends FormControlMixin(RtlMixin(LitElement)) implemen
 
         const componentDisabled = disabled || disabledByParent;
 
+        const checkboxClasses = {
+            'c-checkbox': true,
+            [`c-checkbox--status-${status}`]: !componentDisabled && status,
```
Comment (xander-marjoram) [inline packages/components/pie-checkbox/src/index.ts:170]:
```suggestion
            [`c-checkbox--status-${status}`]: !componentDisabled,
```

File: packages/components/pie-checkbox/test/component/pie-checkbox.spec.ts (hunk 1 above)
Comment (jamieomaguire) [reply packages/components/pie-checkbox/test/component/pie-checkbox.spec.ts:333]: I'm unsure what benefit we think this test provides given we visually test it looks correct when no status is provided to the checkbox. Is there a reason we want to test this attribute @kevinrodrigues?

**Review by xander-marjoram — APPROVED** [review]

File: packages/components/pie-checkbox/test/component/pie-checkbox.spec.ts (hunk 1 above)
Comment (kevinrodrigues) [reply packages/components/pie-checkbox/test/component/pie-checkbox.spec.ts:333]: Im ok with it being deleted. I do see the benefit of having this test tho.

**Review by kevinrodrigues — APPROVED** [review]

**Pull request merged**